src/CachePeer.h

```diff
@@ -48,6 +48,12 @@ class CachePeer
     /// \returns the effective connect timeout for the given peer
     time_t connectTimeout() const;
 
+    /// output to stream a representation of the peer's statistics
+    void reportStatistics (std::ostream& s);
+
+    /// output to stream a representation of the peer's configuration options
+    void dumpOptions(std::ostream& );
```

Comment:
```suggestion
    void dumpOptions(std::ostream &) const;
```

And please check all other new methods for const-correctness.